presidio-analyzer/analyzer/analyzer_engine.py

```diff
@@ -1,47 +1,54 @@
-import logging
-import os
+import uuid
 
 import analyze_pb2
 import analyze_pb2_grpc
 import common_pb2
 
 from analyzer import RecognizerRegistry
+from analyzer.logger import Logger
+from analyzer.app_tracer import AppTracer
 from analyzer.nlp_engine import SpacyNlpEngine
 
-loglevel = os.environ.get("LOG_LEVEL", "INFO")
-logging.basicConfig(
-    format='%(asctime)s:%(levelname)s:%(message)s', level=loglevel)
 
 DEFAULT_LANGUAGE = "en"
+logger = Logger()
 
 
 class AnalyzerEngine(analyze_pb2_grpc.AnalyzeServiceServicer):
 
     def __init__(self, registry=RecognizerRegistry(),
-                 nlp_engine=SpacyNlpEngine()):
+                 nlp_engine=SpacyNlpEngine(),
+                 ml_tracer=AppTracer()):
         # load nlp module
         self.nlp_engine = nlp_engine
         # prepare registry
         self.registry = registry
         # load all recognizers
         registry.load_predefined_recognizers()
+        self.ml_tracer = ml_tracer
 
     # pylint: disable=unused-argument
     def Apply(self, request, context):
-        logging.info("Starting Apply")
+        logger.info("Starting Analyzer's Apply")
+
         entities = AnalyzerEngine.__convert_fields_to_entities(
             request.analyzeTemplate.fields)
         language = AnalyzerEngine.get_language_from_request(request)
-        results = self.analyze(request.text, entities, language,
+
+        generated_request_id = str(uuid.uuid4())
```

Comment:
If we generate it here, it should be called differently. "correlation_id" or something like that. It's not a request Id because it doesn't come with the request

Reply:
This is a temporal fix. It should be recieved from the API. not generated in the analyzer. 
we wanted to minimize the scope of this change, and later on to change it so the ID is generated by the API.

Reply:
Regardless of this being temporary or not, it's confusing to call it "generated request id". Even if we plan to refactor it at a later phase, it should still be named properly, which is a correlation_id or input_id or something in that nature.